.github/workflows/validate.yml

```diff
@@ -0,0 +1,49 @@
+name: validate
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - main
+  pull_request:
+    branches:
+      - master
+      - main
+
+jobs:
+  linters:
+    permissions:
+      contents: read      # for actions/checkout to fetch code
+      pull-requests: read # for golangci/golangci-lint-action to fetch pull requests
+    timeout-minutes: 10
+
+    strategy:
+      matrix:
+        # We only run on the latest version of go, as some linters may be
+        # version-dependent (for example gofmt can change between releases).
+        go-version: [1.19.x]
+        os: [ubuntu-22.04, windows-2019]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+          skip-cache: true
+          args: --print-resources-usage --timeout=5m --verbose
```

Comment:
You can provide configuration through .golangci.yml , it is automatically used by golangci/golangci-lint-action

Reply:
I was on the fence wether to already add one; this was mostly to set up an initial action (using mostly the defaults), and to go from there. Happy to already add a `.golangci.yml` though.

Reply:
Even if it is a really small file at the beginning, It already separates the concerns of lint action (workflow file), and linters configuration

Reply:
Makes sense; updated 👍

Reply:
Thanks, I also prefer splitting out linter config... it tends to grow over time. 😁